themes/src/main/resources/theme/keycloak.v2/login/register.ftl

```diff
@@ -47,7 +47,7 @@
         </form>
 
         <template id="errorTemplate">
-            <div class="${properties.kcFormHelperTextClass}" aria-live="polite">
+            <div id="password-error-container" class="${properties.kcFormHelperTextClass}" aria-live="polite">
```

Comment:
Rather than tracking this element by `id` which could result in duplicate elements with the same ID, it would be better to keep track of the inserted elements in JavaScript and remove them there.

Reply:
I removed the id and instead used `document.getElementById("input-error-client-password").firstElementChild`.

Is this okay or do you really want each message added/removed instead of re-setting the whole error div?

Reply:
I realize now that we can do this slightly different. Instead of pruning the elements from the DOM we can replace them all in one go using [`replaceChildren()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/replaceChildren). So this could be:

```js
document.getElementById("input-error-client-password").replaceChildren(template);
```

Reply:
You're right, this solves the problem and is way cleaner.